design/controlling-agent-updates.md

```diff
@@ -0,0 +1,85 @@
+If a selected agent doesn't meet that requirement, the agent is first asked to update itself.
+Other ways an agent can be updated include using a feature which demands an evergreen agent (YAML pipelines, for instance) or clicking the "upgrade all agents" button in a pool.
+
+For some highly-regulated customers, this kind of evergreen infrastructure is at odds with their security/compliance needs.
+- For instance, Customer "K" runs the agent inside a highly change-controlled datacenter.
+Every alteration must be approved by a high-ranking official in their organization after manual source inspection.
+Today, they diff between the release tags of their current version and the version being installed, looking for potential security issues or malicious changes.
+- Customer "N" isn't as concerned about the security aspect but is in a regulated space.
+They must satisfy their auditors that they're aware of all changes taking place inside their datacenter.
+
+Such customers today are doing things like ACLing the agent's installation directory so it can't update itself or recompiling the agent without update support.
+This leaves us in a situation where no one is happy:
+we can't service the product correctly, and the customer has to maintain custom configuration / infrastructure to meet their compliance needs.
+
+## Solution
+
+_See below for other solutions considered._
+
+We will add a mode to the organization-level pool which puts it into "no update" mode.
+In this mode, jobs will use the minimum agent version as a demand.
+If no agent can satisfy the demand, the job will fail instead of forcing an agent update.
+
+The error message must clearly indicate what feature or task demanded agent update.
+It must also indicate that the customer opted into this behavior.
+This way, the customer can decide whether to update their agents or edit the pipeline to remove the offending feature.
+For example:
+> This pool doesn't allow agent updates.
+> An administrator can alter this setting on the organization pool page.
+> Task `MyTask` version 1.2.3 requires agent version 2.175.1, which could not be satisfied.
+
+Manual agent updates (i.e. clicking the button in the web) should still send update requests as normal.
+
+Additionally, the pool UI will be updated to show the agent's version.
+Any agents not fully up to date (as compared to the version on the AT) should be visually marked.
+(Not being 100% up to date isn't necessarily a problem, so it should info-level, not warning-level.)
+
+## Potential problems & solutions
+
+**If a small number of agents have been updated, then all jobs will funnel to that agent instead of spreading out across the pool.**
+This behavior must be well-documented, both in user-facing material and in troubleshooting guides for the team.
+We will inevitably see instances where this setting is on and causing a pool of agents to sit idle while a single agent in the pool is backed up.
```

Comment:
It would be cool to have a panel in the pool info where customers could put a custom message and we could draw something like "version upgrade in progress"